### PR TITLE
Benchmark CI (actual)

### PR DESCRIPTION
**Pull request by vwxyzjn**
Closes #753. Have to be an internal branch (fork does not work https://github.com/xt0rted/pull-request-comment-branch/issues/385)

### CODE REVIEW

**Comment by vwxyzjn** [comment]
Got
```
Running task 0 with seed: 1
Traceback (most recent call last):
  File "examples/scripts/sentiment_tuning.py", line 20, in <module>
    from accelerate import Accelerator
  File "/admin/home/costa/.local/lib/python3.8/site-packages/accelerate/__init__.py", line 3, in <module>
    from .accelerator import Accelerator
  File "/admin/home/costa/.local/lib/python3.8/site-packages/accelerate/accelerator.py", line 41, in <module>
    from .tracking import LOGGER_TYPE_TO_CLASS, GeneralTracker, filter_trackers
  File "/admin/home/costa/.local/lib/python3.8/site-packages/accelerate/tracking.py", line 60, in <module>
    from aim import Run
  File "/admin/home/costa/.local/lib/python3.8/site-packages/aim/__init__.py", line 2, in <module>
    from aim.sdk import *
  File "/admin/home/costa/.local/lib/python3.8/site-packages/aim/sdk/__init__.py", line 2, in <module>
    from aim.sdk.legacy.session import Session
  File "/admin/home/costa/.local/lib/python3.8/site-packages/aim/sdk/legacy/session/__init__.py", line 1, in <module>
    from aim.sdk.legacy.session.session import Session, DefaultSession
  File "/admin/home/costa/.local/lib/python3.8/site-packages/aim/sdk/legacy/session/session.py", line 12, in <module>
    from aim.sdk.repo import Repo
  File "/admin/home/costa/.local/lib/python3.8/site-packages/aim/sdk/repo.py", line 19, in <module>
    from aim.sdk.run import Run
  File "/admin/home/costa/.local/lib/python3.8/site-packages/aim/sdk/run.py", line 19, in <module>
    from aim.sdk.sequence_collection import SingleRunSequenceCollection
  File "/admin/home/costa/.local/lib/python3.8/site-packages/aim/sdk/sequence_collection.py", line 9, in <module>
    from aim.sdk.query_utils import RunView, SequenceView
  File "/admin/home/costa/.local/lib/python3.8/site-packages/aim/sdk/query_utils.py", line 12, in <module>
    from aim.storage.structured.sql_engine.entities import ModelMappedRun
  File "/admin/home/costa/.local/lib/python3.8/site-packages/aim/storage/structured/sql_engine/entities.py", line 4, in <module>
    from sqlalchemy.orm import joinedload
ModuleNotFoundError: No module named 'sqlalchemy'
srun: error: ip-26-0-145-152: task 0: Exited with exit code 1
```

adding `sqlalchemy` to test dependency

**Comment by vwxyzjn** [comment]
**[COSTA BENCHMARK BOT]**: Here are the benchmark results

**Comment by vwxyzjn** [comment]
**[COSTA BENCHMARK BOT]**: Here are the benchmark results

**Comment by vwxyzjn** [comment]
**[COSTA BENCHMARK BOT]**: Here are the benchmark results

**Comment by vwxyzjn** [comment]
**[COSTA BENCHMARK BOT]**: Here are the benchmark results

**Comment by vwxyzjn** [comment]
**[COSTA BENCHMARK BOT]**: Here are the results

**Comment by vwxyzjn** [comment]
Alright it works! @younesbelkada could you help review?

**Review by younesbelkada — COMMENTED** [review]
Thanks a lot for your great work and the very elegant way of benchmarking incoming PR ! 
I left two comments, regarding running tests from forks, can't we create an additional workflow that acts as follows
1- check if the PR comes from a fork
2- Create a copy of the branch on a local branch in TRL, let's say I have created a branch `dummy-bench` and we want to move it in a branch named `temp-bench-pr-xxx` with `xxx` being the PR number

```bash
git fetch https://github.com/younesbelkada/trl.git dummy-bench:temp-bench-xxx
git checkout temp-bench-xxx
git push origin temp-bench-xxx
```
Then trigger the workflow from that branch and report the results in the PR. What do you think?

File: benchmark/benchmark_core.sh [hunk 1]
```diff
@@ -11,14 +11,14 @@ python benchmark/benchmark.py \
...
+# python benchmark/benchmark.py \
+#     --command "python examples/scripts/sentiment_tuning.py --ppo_config.exp_name sentiment_tuning_gpt2xl_grad_accu --ppo_config.model_name gpt2-xl --ppo_config.mini_batch_size 16 --ppo_config.gradient_accumulation_steps 8 --ppo_config.log_with wandb" \
+#     --num-seeds 3 \
+#     --start-seed 1 \
+#     --workers 10 \
+#     --slurm-nodes 1 \
+#     --slurm-gpus-per-task 1 \
+#     --slurm-ntasks 1 \
+#     --slurm-total-cpus 12 \
+#     --slurm-template-path benchmark/trl.slurm_template
```
[the remark this answers is not in the document]
Comment (vwxyzjn) [reply benchmark/benchmark_core.sh:24]: Done

File: benchmark/benchmark_core.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # hello world experiment
 python benchmark/benchmark.py \
     --command "python examples/scripts/sentiment_tuning.py --ppo_config.log_with wandb" \
-    --num-seeds 5 \
+    --num-seeds 3 \
     --start-seed 1 \
     --workers 10 \
     --slurm-nodes 1 \
```
[the remark this answers is not in the document]
Comment (vwxyzjn) [reply benchmark/benchmark_core.sh:8]: We probably want to keep it as it is. The reason is that `benchmark/benchmark_core.sh` can be standalone for people who just want to run it without Github actions.

[the remark this answers is not in the document]
Comment (younesbelkada) [reply benchmark/benchmark_core.sh:8]: Sounds great to me!

**Review by younesbelkada — APPROVED** [review]
Thanks very much again for your great work on this!

**Pull request merged**